nntrainer/src/neuralnet.cpp

```diff
@@ -344,6 +344,7 @@ void NeuralNetwork::saveModel() {
   std::ofstream model_file(save_path, std::ios::out | std::ios::binary);
   for (unsigned int i = 0; i < layers.size(); i++)
     layers[i]->save(model_file);
+  model_file.write((char *)&epoch_idx, sizeof(epoch_idx));
```

Comment:
do we really need to store/load both epoch_idx and iter?
epoch_idx can be derived from iter right?

Reply:
I think epoch_idx can be local to `train_run` function and be locally derived there itself.

Reply:
Can we calculate epoch_idx when we are using `dataset_with_generator`?
I think we can calculate epoch_idx iff `iteration per epoch` is known to the `neuralnet`.

Reply:
May be cache iteration per epoch after the epoch then (initialized to 0)?
Then epoch_idx can be inferred, and it can be further used later to display progress too.